core/web-assets/src/main/assets/js/apps/onms-classifications/index.js

```diff
@@ -312,6 +314,59 @@ const confirmTopoverTemplate = require('./views/modals/popover.html');
                 });
             };
 
+            // for drag and drop of rules (redefining position)
+            $scope.sortableRules = {
+
+                start: function(e, ui) {
+                    // remember old index before moving
+                    $(ui.item).data('oldIndex', ui.item.index());
+                },
+                stop: function(e, ui) {
+
+                    // 1.) Check Precondition: not read only group
+                    if($scope.group.readOnly) {
+                        ui.item.parent().sortable('cancel');
+                        $window.alert('this group cannot be sorted, it is read only');
+                        return;
+                    }
+
+                    // 2.) Check Precondition: must be sorted by position
+                    if(!($scope.query.orderBy === 'position' && $scope.query.order === 'asc')) {
```

Comment:
Ideally we disable re-arranging when not sorted by position

Reply:
fixed.